attestation-agent/attester/src/lib.rs

```diff
@@ -157,3 +162,162 @@ pub fn detect_tee_type() -> Tee {
     );
     Tee::Sample
 }
+
+/// Get any additional TEEs that might be connected to the guest,
+/// such as a confidential device.
+pub fn detect_attestable_devices() -> Vec<Tee> {
+    // TODO
+    vec![]
+}
+
+/// The Attesters struct is an interface to all the attesters
+/// that represent a confidential guest.
+pub struct Attesters {
+    primary_attester_type: Tee,
+    primary_attester: BoxedAttester,
+    additional_attesters: HashMap<Tee, BoxedAttester>,
+}
+
+/// GuestEvidence is the combined evidence from all the TEEs
+/// that represent the guest.
+#[derive(Serialize, Deserialize)]
+pub struct GuestEvidence {
+    primary_evidence: String,
+    // The additional evidence is a map of Tee -> evidence,
+    // but we convert it to a string to avoid any inconsistencies
+    // with serialization. The string in this struct is exactly
+    // what is used to calculate the runtime data.
+    additional_evidence: String,
```

Comment:
In the serde_json crate, the serialization result is partially [canonicalized](https://www.rfc-editor.org/rfc/rfc8785) (in cases where only strings and literals are used), so if we do not use floats it might be enough to use `serde_json::Value` here?

Reply:
I went back and forth on this a little bit. I think it can work with a `HashMap` or a `BTreeMap` here, but with a String there might be more peace of mind. You don't have to spend any time thinking about whether it will work.

Also, note that I changed the return type of the attesters to be a Value. If we have a map here, we would probably want to change that Value back to a string to make sure we don't have any serialization inconsistencies with the evidence itself, which is totally arbitrary.

Anyway, I opted to make the evidence a value but turn the map of evidence into a String, but I am open to other ideas.

Reply:
Yup. String is always safe. Crates like https://crates.io/crates/olpc-cjson can make the serialization consistent but imports more dependency.